### DISCUSSION TITLE
New RNode display features

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#144] [Show and tell]
Hi everyone, here's a little preview of some stuff I have been working on:

![rnfw_1](https://user-images.githubusercontent.com/1438012/198842158-cf7ed04d-e477-400b-914b-8d0a0c68f426.jpg)

![rnfw_2](https://user-images.githubusercontent.com/1438012/198842160-440f8b6c-725a-48e1-b873-de26142c6589.jpg)

The display now shows:

- Transceiver status
- Sub-transceiver status for LoRa and 2.4 gig radios
- Cable (USB/serial) connection status
- Bluetooth status
- Battery status (if supported by hardware)
- A RSSI graph of the tuned channel, letting you quickly see activity and noise levels of the spectrum (updates in real-time)
- Signal strength and signal quality bar graphs
- Better diagnostics info on hardware errors and such

The display is divided into two areas, one is a status area that is always managed by the firmware itself, with the above info.
The other is a flexible area that can be used by the firmware to display various info, or can be put into "external framebuffer mode" and controlled directly by the connected software.

In the pictures above, Sideband is connected to one of the RNodes, and it is driving the display to show a Sideband logo. Programs can do full or partial updates to the framebuffer, so it can be used to quickly display all kinds of info, like message notifications or even message text. Anything really :)

I especially like the "waterfall" graph showing channel activity, it really makes it easy to get an idea about what is going on, both in terms of activity and noise levels. You will notice one of the RNodes shows a very high noise floor in the 430 MHz band, while the other one shows a much more quiet environment in the 900 MHz band.

**Comment by stevewa2066** [comment]
Maybe on a different screen : name/call , nodes connected.

**Comment by kc1awv** [comment]
I just wanted to show off. The new display looks awesome!
![66CE9B2B-C4EA-4E0B-90B8-74A8F7CE6F7D](https://user-images.githubusercontent.com/43918257/200838053-e037fe50-bc2c-4a27-b25e-997db98c150c.jpeg)

**Reply by markqvist** [reply]
Very nice! That is looking great!

**Comment by stevewa2066** [comment]
Great!

Steve N0FPF

**Comment by dd7rg** [comment]
I just started using Reticulum and RNodes and I like it very much. One thing which came to mind is to have a sort of display saver or "covert" mode, like on tetra radios ;-) ... On the one hand for saving the oled and maybe one may not get to much attention of rnodes lying around. Just have a look a the code how this can be done without a button. Maybe a config variable which can be written by nomad/sideband or similar (?)
Again thanks a lot for creating this wonderful ecosystem 
Ralf

**Reply by markqvist** [reply]
I can't remember if I already added a hidden option to `rnodeconf` for setting the display brightness (including turning it off completely). The function is already there in the firmware at least :) I will make sure to add it in the next release (or so).

**Reply by dd7rg** [reply]
Would be appreciated :-)

**Reply by pizzadoglover911** [reply]
It does not appear like there is an option in rnodeconf, at least accessible through the normal runtime. Has this been added or is there somewhere I can find it?

I would like to reduce power consumption and accurately size batteries/solar panels. Thanks!

**Reply by markqvist** [reply]
I'll make sure it is added for the next release :)
